modules/sdk-coin-sui/src/lib/stakingBuilder.ts

```diff
@@ -0,0 +1,162 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { InvalidTransactionError, NotImplementedError, TransactionType } from '@bitgo/sdk-core';
+import {
+  MoveCallTx,
+  MethodNames,
+  ModulesNames,
+  MoveCallTxDetails,
+  RequestAddDelegationTx,
+  SuiTransaction,
+  SuiTransactionType,
+  SuiObjectRef,
+} from './iface';
+import { TransactionBuilder } from './transactionBuilder';
+import { SUI_PACKAGE, SUI_SYSTEM_STATE_OBJECT } from './constants';
+import { SuiMoveCallTransactionSchema } from './txnSchema';
+import { StakingTransaction } from './stakingTransaction';
+import { Transaction } from './transaction';
+import BigNumber from 'bignumber.js';
+
+export class StakingBuilder extends TransactionBuilder<MoveCallTx> {
+  protected _moveCallTx: MoveCallTx;
+  protected _addDelegationTx: RequestAddDelegationTx;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._transaction = new StakingTransaction(_coinConfig);
+  }
+
+  /**
+   * Build a MoveCall transaction ready to be signed and executed.
+   *
+   * @returns {UnserializedSignableTransaction} an unsigned Sui transaction
+   *
+   * @see https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/docs/sui_system.md#function-request_add_delegation_mul_coin
+   */
+  protected buildStakeTransaction(): SuiTransaction<MoveCallTx> {
+    return {
+      type: SuiTransactionType.AddDelegation,
+      sender: this._sender,
+      tx: this._moveCallTx,
+      gasBudget: this._gasBudget,
+      gasPrice: this._gasPrice,
+      gasPayment: this._gasPayment,
+    };
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.AddDelegator;
+  }
+
+  /**
+   * Create a new transaction for delegating coins ready to be signed and executed.
+   *
+   * @param addDelegationTx
```

Comment:
```suggestion
   * @param {RequestAddDelegationTx} addDelegationTx
```
Lets document code consistently